test/integration/instance.test.js

```diff
@@ -898,6 +898,38 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
       });
     });
 
+    it('handles an entry with primaryKey of zero', function() {
+      var username = 'user'
+        , newUsername = 'newUser'
+        , User2 = this.sequelize.define('User2',
+          {
+            id: {
+              type: DataTypes.INTEGER.UNSIGNED,
```

Comment:
Seems like you found a bug ^^. Mssql does not support unsigned integer. I'll fix that later.